src/proxyvisor/proxyvisor.coffee

```diff
@@ -471,7 +274,7 @@ module.exports = class Proxyvisor
 					config: device.apps[appId].config
 					environment: device.apps[appId].environment
 				}
-				if device.targetCommit? and !_.isEqual(targetState, currentState) and !_.isEqual(targetState, @acknowledgedState[device.uuid])
+				if device.apps[appId].targetCommit? and !_.isEqual(targetState, currentState) and !_.isEqual(targetState, @acknowledgedState[device.uuid])
```

Comment:
Nice catch here :) I had found it when debugging on your device but forgot to change it